hashicorp/dev-portal
# feat: update enterprise alert

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-leah-featupdate-enterprise-alert-hashicorp.vercel.app/hcp/docs/boundary/self-managed-workers/session-recording) 🔎
- [Asana task](https://app.asana.com/0/1208691590278290/1208502489898170/f) 🎟️

## 🗒️ What


This PR updates the `EnterpriseAlert` component to support `vault-secrets` and `vault-radar`. It also adds the `badgeText` prop so a custom message can be shown in the box instead of just 'Enterprise'. 

## 🤷 Why


This feature came from a [support request](https://hashicorp.slack.com/archives/C5FSPUGDS/p1728322562399209) in the team-web-support slack channel.

## 📸 Design Screenshots


Vault secrets
<img width="707" alt="Screenshot 2024-11-06 at 3 04 48 PM" src="https://github.com/user-attachments/assets/abd2c37e-cd95-4ae6-a1a9-05b2a243ff3d">
Vault radar
<img width="706" alt="Screenshot 2024-11-06 at 3 10 59 PM" src="https://github.com/user-attachments/assets/1e1ad7cc-b7ed-4957-ac2e-4d93c3e52244">



## 🧪 Testing



1. Checkout this branch 
```
git checkout leah/feat/update-enterprise-alert
``` 
2. Start the app locally
3. Go to [local link](http://localhost:3000/hcp/docs/boundary/self-managed-workers/session-recording)
4. Open dev tools
5. In the components tab, use the select tool to click on the `DocsEnterpriseAlert` component
6. In the component props do the following three things: 
- Add the `badgeText` prop with a custom string
- Change the `product` prop to `vault-secrets` or `vault-radar`
- Delete the children prop
7. Click enter
8. Verify that the correct icon is shown for secrets or radar, the badgeText string passed is shown in the box, and that the link brings you to the corresponding vault product page
